econml/bootstrap.py

```diff
@@ -212,21 +220,31 @@ def get_dist(est, arr):
                         return 2 * est - arr
                     else:
                         raise ValueError("Invalid kind, must be either 'percentile' or 'pivot'")
-                return proxy(callable(getattr(self._instances[0], prefix)), prefix,
-                             lambda arr, est: EmpiricalInferenceResults(d_t=d_t, d_y=d_y,
-                                                                        pred=est, pred_dist=get_dist(est, arr),
-                                                                        inf_type=inf_type, fname_transformer=None))
 
-            def get_inference_parametric():
+                result = proxy(can_call, prefix,
+                               lambda arr, est: EmpiricalInferenceResults(d_t=d_t, d_y=d_y,
+                                                                          pred=est, pred_dist=get_dist(est, arr),
+                                                                          inf_type=inf_type,
+                                                                          fname_transformer=fname_transformer))
+                # Note that inference results are always methods even if the inference is for a property
+                # (e.g. coef__inference() is a method but coef_ is a property)
+                # Therefore we must insert a lambda if getting inference for a non-callable
+                return result if can_call else lambda: result
+
+            def get_inference_parametric(*args, **kwargs):
                 pred = getattr(self._wrapped, prefix)
+                if can_call:
+                    pred = pred(*args, **kwargs)
                 stderr = getattr(self, prefix + '_std')
+                if can_call:
+                    stderr = stderr(*args, **kwargs)
                 return NormalInferenceResults(d_t=d_t, d_y=d_y, pred=pred,
                                               pred_stderr=stderr, inf_type=inf_type,
-                                              pred_dist=None, fname_transformer=None)
+                                              fname_transformer=fname_transformer)
 
-            return {'normal': get_inference_parametric,
-                    'percentile': lambda: get_inference_nonparametric('percentile'),
-                    'pivot': lambda: get_inference_nonparametric('pivot')}[self._bootstrap_type]
+            return {'normal': get_inference_parametric if can_call else lambda: get_inference_parametric(),
+                    'percentile': get_inference_nonparametric('percentile'),
+                    'pivot': get_inference_nonparametric('pivot')}[self._bootstrap_type]
 
         caught = None
         m = None
```

Comment:
Currently no test reaches the branch where self._compute_means=True and self._prefer_wrapped=True.

Do we want to maintain these flags? If so then what functionality are they serving and if they are serving some functionality we should add a test that verifies it. Alternatively we can simply remove these flags and remove this branch from here if that's the only place they are used, just for the shake of maintaining a clean code.

Reply:
Removed that code since it added complexity for no benefit.